ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/LedgerStateAccess.scala

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2020 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.validator
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait LedgerStateAccess {
+  def inTransaction[T](body: LedgerStateOperations => Future[T]): Future[T]
+
+  def participantId: String
+}
+
+trait LedgerStateOperations {
+  def readState(key: Array[Byte]): Future[Option[Array[Byte]]]
+  def readState(keys: Seq[Array[Byte]]): Future[Seq[Option[Array[Byte]]]]
```

Comment:
Can we get some type aliases in here so I don't have to think so hard?

Perhaps:

```scala
type Key = Array[Byte]
type Value = Option[Array[Byte]]

// ...

def readState(key: Key): Future[Value]
def readState(keys: Seq[Key]): Future[Seq[Value]]
```

Reply:
And perhaps some Scaladoc, especially for pointing out that the batching read _must_ return values in the same order as the keys.

Reply:
Or you could have `readState` return a `Future[Map[Key, Value]]`, which you're constructing anyway; it'd probably be just as easy for the implementer to handle this and then they don't have to worry about order.

Reply:
Good point -- added Key & Value types.

Reply:
I'll change the signature of `readState` to return a `Map` instead of a `Seq`.

Reply:
Added a bit of Scaladocs -- @SamirTalwar-DA please let me know what you think.